File: result/repair/transcoder/java2cpp/c++_repair/MINIMUM_TIME_WRITE_CHARACTERS_USING_INSERT_DELETE_COPY_OPERATION.cpp
int f_gold ( int N , int insert , int remove , int copy ) {
  if ( N == 0 ) return 0 ;
  if ( N == 1 ) return insert ;
  int dp [ N + 1 ] = {0} ;
memset(dp,0,sizeof(dp));
  for ( int i = 1 ;
  i <= N ;
  i ++ ) {
    if ( i % 2 == 0 ) {
      dp [ i ] = min ( dp [ i - 1 ] + insert , dp [ i / 2 ] + copy ) ;
    }
    else {
      dp [ i ] = min ( dp [ i - 1 ] + insert , dp [ ( i + 1 ) / 2 ] + copy + remove ) ;
    }
  }
  return dp [ N ] ;
}

File: dataset/java2cpp2(BatFix)/java/MINIMUM_TIME_WRITE_CHARACTERS_USING_INSERT_DELETE_COPY_OPERATION.java
int f_gold ( int N, int insert, int remove, int copy ) {
  if ( N == 0 ) return 0;
  if ( N == 1 ) return insert;
  int dp [ ] = new int [ N + 1 ];
  for ( int i = 1;
  i <= N;
  i ++ ) {
    if ( i % 2 == 0 ) dp [ i ] = Math . min ( dp [ i - 1 ] + insert, dp [ i / 2 ] + copy );
    else dp [ i ] = Math . min ( dp [ i - 1 ] + insert, dp [ ( i + 1 ) / 2 ] + copy + remove );
  }
  return dp [ N ];
}